### ISSUE TITLE
`role="button"` not needed when using `<button>` elements

### ISSUE THREAD

**Issue by ayumi-cloud** [ampproject/amphtml#31342]
Doc's code example 1, link here:

https://amp.dev/documentation/components/amp-consent/#consent-configuration

```html
<amp-consent layout="nodisplay" id="consent-element">
  <script type="application/json">
    {
      "consentInstanceId": "my-consent",
      "consentRequired": "remote",
      "checkConsentHref": "https://example.com/api/check-consent",
      "promptUI": "consent-ui",
      "onUpdateHref": "https://example.com/update-consent"
    }
  </script>
  <div id="consent-ui">
    <button on="tap:consent-element.accept" role="button">Accept</button>
    <button on="tap:consent-element.reject" role="button">Reject</button>
    <button on="tap:consent-element.dismiss" role="button">Dismiss</button>
  </div>
</amp-consent>
```

- You don't need to add `role="button"` on a `<button>` element.

Doc's code example 2, link here:

https://amp.dev/documentation/components/amp-consent/#prompt-ui

```html
<amp-consent layout="nodisplay" id="consent-element">
  <script type="application/json">
    {
      "consentInstanceId": "my-consent",
      "checkConsentHref": "https://foo.com/api/show-consent",
      "promptUI": "consent-ui"
    }
  </script>
  <div id="consent-ui">
    <button on="tap:consent-element.accept" role="button">Accept</button>
    <button on="tap:consent-element.reject" role="button">Reject</button>
    <button on="tap:consent-element.dismiss" role="button">Dismiss</button>
  </div>
</amp-consent>
```

- You don't need to add `role="button"` on a `<button>` element.

---

### Correct method

https://github.com/ampproject/amp.dev/blob/3ced33a2c431cb03c41244613baadc3179692467/examples/source/user-consent/Geolocation-based_Consent_Flow.html#L160

```html
<div class="dismiss-button" role="button" tabindex="0" on="tap:myUserConsent.dismiss">X</div>
```

and

```html
<button on="tap:consent-element.accept">Accept</button>
```

### Incorrect method

```html
<div class="dismiss-button" tabindex="0" on="tap:myUserConsent.dismiss">X</div>
```

and

```html
<button on="tap:consent-element.accept" role="button">Accept</button>
```

---

Done a check in this repo and this mistake exists in many examples and extensions!

For example:
https://github.com/ampproject/amphtml/blob/1a3b14932be40b2370f6726494295963fc7b9532/examples/gwd.amp.html#L103

I suggest the authors go through this repo and correct them all as well.

The same mistake is seen in this repo as well: https://github.com/ampproject/amp.dev

For example:
https://github.com/ampproject/amp.dev/blob/4f12d7c2443ddf6fc75eaecf8bf53ac7e08fffd1/frontend/templates/views/partials/fragment-slider-controls.j2#L19

**Comment by nickFalcone** [comment]
If helpful, I can open a pull request to address instances within amp-consent examples and documentation?

I searched the whole repo for `setAttribute('role', 'button')` and found 8 occurrences. Some if not all of these are likely valid.

@krdwan - do you know if there is a good place to add a global test for presence of `role` attribute on a `<button>` ? Something like:

```js
[...document.getElementsByTagName('BUTTON')].forEach((button) => {
    expect(button.hasAttribute('role')).to.be.false;
});
```

**Comment by kristoferbaxter** [comment]
Resolved with #32252.
